src/sentry/snuba/metrics/query.py

```diff
@@ -290,6 +290,8 @@ def action_by_str_fields(self, on_group_by: bool) -> Set[str]:
     def validate_limit(self) -> None:
         if self.limit is None:
             return
+        if self.start is None or self.end is None:
+            raise Exception("We need start and end to be set.")
```

Comment:
`start` and `end` are marked as `Optional[datetime]` but the function below requires both values.